R/validate_vpts.R

```diff
@@ -0,0 +1,107 @@
+vpts_schema <- get0("vpts_schema", envir = asNamespace("bioRad"))
+
+get_field_schema <- function(field, schema) {
+    for (i in seq_along(schema$fields$name)) {
+        if (field %in% c(schema$fields$name[i], schema$fields$nameAlternatives[[i]])) {
+            return(schema$fields[i, ])
+        }
+    }
+    return(NULL)  # return NULL if no matching field found
+}
+validate_datetime_format <- function(data, format) {
+  parsed_data <- tryCatch({
+    as.POSIXct(data, format = format, tz = "UTC")
+  }, error = function(e) NULL)
+  # check for failed parsing
+  if (any(is.na(parsed_data))) {
+    return(FALSE)
+  }
+  return(TRUE)
+}
+#' Validate dataframe against VPTS schema
+#'
+#' @param df The dataframe to validate.
+#'
+#' @return Invisibly returns a list with validation results including messages for any issues found.
+#' @importFrom glue glue
+#' @importFrom dplyr select
+#' @importFrom stringr str_detect
+#' @keywords internal
+#' @examples
+#' my_vpts <- as.data.frame(example_vpts)
```

Comment:
The warning message I get with this example seems to be incorrect, it lists all the available fields, not only the ones that were additional:

```
> validate_vpts(my_vpts)
Warning message:
In validate_vpts(my_vpts) :
  Extra fields found: radar, datetime, height, u, v, w, ff, dd, sd_vvp, gap, dbz, eta, dens, DBZH, n, n_dbz, n_all, n_dbz_all, rcs, sd_vvp_threshold, radar_latitude, radar_longitude, radar_height, radar_wavelength, day, sunrise, sunset
  ```
 Also, note you can turn off some of the additional fields, include examples for these additional flavors?
 ```
as.data.frame(example_vpts, geo=TRUE, suntime=FALSE)
as.data.frame(example_vpts, geo=FALSE, suntime=TRUE)
as.data.frame(example_vpts, geo=FALSE, suntime=FALSE)

 ```

Reply:
`geo` should never be False as that would remove required VPTS fields, which include:

```
radar, datetime, height, radar_latitude, radar_longitude, radar_height, radar_wavelength
```
we might consider removing that parameter